mockrxandroidble/src/main/java/com/polidea/rxandroidble/mockrxandroidble/RxBleClientMock.java

```diff
@@ -0,0 +1,281 @@
+         */
+        public CharacteristicsBuilder() {
+            this.bluetoothGattCharacteristics = new ArrayList<>();
+        }
+
+        /**
+         * Adds a {@link BluetoothGattCharacteristic} with specified parameters.
+         *
+         * @param uuid        characteristic UUID
+         * @param data        locally stored value of the characteristic
+         * @param descriptors list of characteristic descriptors. Use {@link DescriptorsBuilder} to create them.
+         */
+        public CharacteristicsBuilder addCharacteristic(@NonNull UUID uuid,
+                                                        @NonNull byte[] data,
+                                                        List<BluetoothGattDescriptor> descriptors) {
+            BluetoothGattCharacteristic characteristic = new BluetoothGattCharacteristic(uuid, 0, 0);
+            for (BluetoothGattDescriptor descriptor : descriptors) {
+                characteristic.addDescriptor(descriptor);
+            }
+            characteristic.setValue(data);
+            this.bluetoothGattCharacteristics.add(characteristic);
+            return this;
+        }
+
+        /**
+         * Create the  {@link List} of {@link BluetoothGattCharacteristic} using the configured values.
+         */
+        public List<BluetoothGattCharacteristic> build() {
+            return bluetoothGattCharacteristics;
+        }
+    }
+
+    public static class DescriptorsBuilder {
+
+        private List<BluetoothGattDescriptor> bluetoothGattDescriptors;
+
+        /**
+         * Build a new {@link BluetoothGattDescriptor} list.
+         * Should be used in pair with {@link CharacteristicsBuilder#addCharacteristic}
+         */
+        public DescriptorsBuilder() {
+            this.bluetoothGattDescriptors = new ArrayList<>();
+        }
+
+        /**
+         * Adds a {@link BluetoothGattDescriptor} with specified parameters.
+         *
+         * @param uuid descriptor UUID
+         * @param data locally stored value of the descriptor
+         */
+        public DescriptorsBuilder addDescriptor(@NonNull UUID uuid, @NonNull byte[] data) {
+            BluetoothGattDescriptor bluetoothGattDescriptor = new BluetoothGattDescriptor(uuid, 0);
+            bluetoothGattDescriptor.setValue(data);
+            bluetoothGattDescriptors.add(bluetoothGattDescriptor);
+            return this;
+        }
+
+        /**
+         * Create the  {@link List} of {@link BluetoothGattDescriptor} using the configured values.
+         */
+        public List<BluetoothGattDescriptor> build() {
+            return bluetoothGattDescriptors;
+        }
+    }
+
+    private final Map<String, RxBleDevice> availableDevices;
+
+    private RxBleClientMock(Builder builder) {
+        availableDevices = builder.availableDevices;
+    }
+
+    @Override
+    public RxBleDevice getBleDevice(@NonNull String macAddress) {
+        return availableDevices.get(macAddress);
```

Comment:
Throw IllegalState or UnupportedOperation if mock is not configured for a given device.